app/models/attraction.rb

```diff
@@ -30,6 +30,8 @@ class Attraction < ApplicationRecord
                                 :regions
 
   def find_or_create_category
+    return if category.present?
```

Comment:
Dadurch ist kein Update der Kategorie mehr möglich, oder?

Reply:
nicht wenn man einen before_create callback daraus macht

Reply:
so wie es jetzt ist wäre ein update nicht möglich.

Reply:
So habe den callback in ein before_create geändert. Ein Update ist also möglich. genauso wie die creation über den category_name, denn bei einem neuen record der noch keine category hat ist category.present? false und damit wird der category_name benutzt um eine neue category zu erzeugen.

Reply:
- Was ist der eigentliche Grund für diese Anpassung?
- Warum ist durch die Veränderung des Zeitpunkts des Callbacks ein Update der Kategorie möglich?

Beispiel: 

```
a = Attraction.create(category_name: "Foo")
a.category = Category<name: "Foo">

# Beispiel mit "before_create"
a.update_attributes(category_name: "Bar")
< a.find_or_create_category not called because of "before_create" >
a.category = Category<name: "Foo">

# Beispiel mit "before_validation"
a.update_attributes(category_name: "Bar")
a.find_or_create_category
=> category.present? => guard clause occurs, no changes in Category
```

in Beiden Fällen läßt sich meiner Meinung nach die Kategorie nicht mehr ändern, sobald sie einmal gesetzt ist, da die `guard clause` das verhindert oder die Methode erste gar nicht mehr aufgerufen wird.

Reply:
Nichts. Das stimmt.

Reply:
Nichts. Das stimmt.